### DISCUSSION TITLE
RendermanForMaya resource pack seemingly doesn't work?

### DISCUSSION THREAD

**Discussion by barelyaiden** [BramStoutProductions/MiEx#96] [Q&A]
Hi! I've been trying to import a world into Maya to render with Renderman, with only the RendermanForMaya resource pack enabled no material imports through the Maya plugin, and if I use it in conjunction with StandardSurfaceForMaya and RendermanForMaya on top of it, it only imports Standard Surface materials. Am I doing something wrong by chance? I'm using Maya 2024.2 and Renderman 26.1

**Comment by BramStout** [comment]
If you aren't using the [Maya import script](https://github.com/BramStoutProductions/MiEx/blob/main/extras/Maya/Maya_MiEx_Importer.py), then please try it with that.

If it still isn't importing in the Renderman materials, could you share the output of Maya's Script Editor? Often that'll have something in it that points to the underlying issue

**Reply by barelyaiden** [reply]
Yeah I am using the Maya import script, I tried just now with a fresh export and still nothing sadly. Here's the log: https://pastebin.com/1HUG6Say

**Reply by BramStout** [reply]
> Yeah I am using the Maya import script, I tried just now with a fresh export and still nothing sadly. Here's the log: https://pastebin.com/1HUG6Say

Hmmm, the log doesn't say anything. Could you share the export itself? So the .usd files but also the file at ends with `_materials.json` since that file contains the actual Renderman and StandardSurface materials

**Reply by barelyaiden** [reply]
> > Yeah I am using the Maya import script, I tried just now with a fresh export and still nothing sadly. Here's the log: https://pastebin.com/1HUG6Say
> 
> Hmmm, the log doesn't say anything. Could you share the export itself? So the .usd files but also the file at ends with `_materials.json` since that file contains the actual Renderman and StandardSurface materials

Here you go! 
[TestWorld.zip](https://github.com/user-attachments/files/16321534/TestWorld.zip)

**Reply by barelyaiden** [reply]
> > > Yeah I am using the Maya import script, I tried just now with a fresh export and still nothing sadly. Here's the log: https://pastebin.com/1HUG6Say
> > 
> > 
> > Hmmm, the log doesn't say anything. Could you share the export itself? So the .usd files but also the file at ends with `_materials.json` since that file contains the actual Renderman and StandardSurface materials
> 
> Here you go! [TestWorld.zip](https://github.com/user-attachments/files/16321534/TestWorld.zip)

I just did some testing, with only the Standard Surface materials it shows up as gray when rendering but with the Renderman materials on top of that it's all textured, so I guess it DOES work? Then why won't the PxrSurface materials show up in the Hypershade am I checking in the wrong place by chance?

**Reply by BramStout** [reply]
> > > > Yeah I am using the Maya import script, I tried just now with a fresh export and still nothing sadly. Here's the log: https://pastebin.com/1HUG6Say
> > > 
> > > 
> > > Hmmm, the log doesn't say anything. Could you share the export itself? So the .usd files but also the file at ends with `_materials.json` since that file contains the actual Renderman and StandardSurface materials
> > 
> > 
> > Here you go! [TestWorld.zip](https://github.com/user-attachments/files/16321534/TestWorld.zip)
> 
> I just did some testing, with only the Standard Surface materials it shows up as gray when rendering but with the Renderman materials on top of that it's all textured, so I guess it DOES work? Then why won't the PxrSurface materials show up in the Hypershade am I checking in the wrong place by chance?

When the Maya import script creates the different shading nodes, it has to tell it whether it's a material node, texture node, or utility node. It figures this out by querying info about the node type which should tell it what that node type is classified as. Unfortunately, RendermanForMaya doesn't properly classify the PxrSurface material node type, causing the import script to create the node as a utility node. This doesn't affect the material itself, just under what tab in the Hypershade it's under. So the PxrSurface materials are under the utility tab. It's something that I still need to fix, just haven't gotten around to it yet, with all of the work that I'm doing on the upcoming 1.7 release of MiEx

**Reply by barelyaiden** [reply]
> > > > > Yeah I am using the Maya import script, I tried just now with a fresh export and still nothing sadly. Here's the log: https://pastebin.com/1HUG6Say
> > > > 
> > > > 
> > > > Hmmm, the log doesn't say anything. Could you share the export itself? So the .usd files but also the file at ends with `_materials.json` since that file contains the actual Renderman and StandardSurface materials
> > > 
> > > 
> > > Here you go! [TestWorld.zip](https://github.com/user-attachments/files/16321534/TestWorld.zip)
> > 
> > 
> > I just did some testing, with only the Standard Surface materials it shows up as gray when rendering but with the Renderman materials on top of that it's all textured, so I guess it DOES work? Then why won't the PxrSurface materials show up in the Hypershade am I checking in the wrong place by chance?
> 
> When the Maya import script creates the different shading nodes, it has to tell it whether it's a material node, texture node, or utility node. It figures this out by querying info about the node type which should tell it what that node type is classified as. Unfortunately, RendermanForMaya doesn't properly classify the PxrSurface material node type, causing the import script to create the node as a utility node. This doesn't affect the material itself, just under what tab in the Hypershade it's under. So the PxrSurface materials are under the utility tab. It's something that I still need to fix, just haven't gotten around to it yet, with all of the work that I'm doing on the upcoming 1.7 release of MiEx

Ahhh I found it! I honestly was not aware of the material separation in Hypershade since I'm still new to Maya. Thank you for your help!
